haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -434,11 +436,79 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _BedrockEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        boto3_import.check()
+
+        # See https://docs.aws.amazon.com/bedrock/latest/userguide/embeddings.html for more details
+        # The maximum input text is 8K tokens and the maximum output vector length is 1536
+        # Bedrock embeddings do not support batch operations
+        self.model: str = "amazon.titan-embed-text-v1"
+        self.aws_config = retriever.aws_config
+        self.client = self.initialize_boto3_client()
+
+    def initialize_boto3_client(self):
+        if self.aws_config:
+            access_key_id = self.aws_config.get("aws_access_key_id")
+            secret_access_key = self.aws_config.get("aws_secret_access_key")
+            region = self.aws_config.get("region")
+            try:
+                return boto3.client(
```

Comment:
Curious on thoughts about this. It would be fine if accessing bedrock-runtime client from outside aws, but what about initializing from within sagemeaker or a lambda that already has access and doesnt need keys to auth eg aws_config would be empty. Just a thought.

Reply:
We should allow creating bedrock runtime without any AWS config for this case. That's totally fine.